docs/developers/contributing/testing.md

```diff
@@ -39,97 +41,133 @@ from unit tests to integration and functional tests.
-placed here.
+These GUI tests are ignored when we run them in the subset of our continuous integration
+workflows. Workflows that run in a "headless" environment (without a Qt backend).
+Testing of core library, or "non-GUI" code, that requires a specific GUI backend are also found in these folders.
 
-The `napari/plugins` folder contains tests related to plugins.
+### napari plugin tests
 
-Pytest fixtures to aid testing live in:
+The `napari/plugins` folder contains most tests related to plugins.
+
+### pytest fixtures
+
+Pytest fixtures are used to set up test state, such as setup and teardown, and provide frequently test data. These
+fixtures reduce repetitive code when writing and running tests. The fixtures can be found in:
 
 * [`napari/conftest.py`](https://github.com/napari/napari/blob/main/napari/conftest.py) -
   available globally to all of `napari`.
 * [`napari/utils/_testsupport.py`](https://github.com/napari/napari/blob/main/napari/utils/_testsupport.py) -
-  available globally to all of `napari` **and** to all tests in the same environment
-  that `napari` is in (as this file is exported).
+  available globally to all of `napari` **and** to all tests in the same virtual environment
+  that `napari` is in (as this `testsupport.py` file is exported).
+
+### `make_napari_viewer` fixture
 
 One often used fixture is `make_napari_viewer`. This fixture can take an argument `show`
 which is either `True` or `False`. In case your test depends on rendering of the viewer,
-it should be set to `True`. This is, for example, the case when testing a screenshot 
+it should be set to `True`. This is, for example, the case when testing a screenshot
 functionality. Otherwise, it is best to set the argument to `False` to prevent the viewer
 from fully rendering.
 
-There are also fixtures for testing the `napari` builtin plugin (provides contributions
-that come builtin with `napari`).
-These live in
+### napari builtin plugin fixtures
+
+napari comes with a number of samples and examples builtin.
+There are also fixtures for testing these `napari` builtin plugins that provide contributions
+that come builtin with `napari`.
+These fixtures are found in
 [`napari_builtins/_tests/conftest.py`](https://github.com/napari/napari/blob/main/napari_builtins/_tests/conftest.py)
-and are available within
+and are available to tests stored in
 [`napari_builtins/_tests`](https://github.com/napari/napari/tree/main/napari_builtins/_tests).
 
 (running-tests)=
 
 ## Running tests
 
-To run our test suite locally, run `pytest` on the command line.  If, for some reason
-you don't already have the test requirements in your environment, run `python -m pip install -e .[testing]`.
+To run our test suite locally, run `pytest` on the command line.
+When running tests, you will want to have development and testing dependencies installed
+in your environment:
+
+```sh
+python -m pip install -e .[dev, testing]
```

Comment:
`dev` includes testing.
What is missing here is a Qt backend.
Maybe just point to setting up a dev install:
https://napari.org/stable/developers/contributing/dev_install.html#setting-up-a-development-installation

Reply:
I've added Peter's suggestion.